geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/CommonParams.scala

```diff
@@ -31,13 +31,17 @@ trait TypeNameParam {
 }
 
 trait RequiredTypeNameParam extends TypeNameParam {
+  private val ESCAPE = "\u0000" // For escaping tilde
   @Parameter(names = Array("-f", "--feature-name"), description = "Simple Feature Type name on which to operate", required = true)
   var featureName: String = null
+  if (featureName != null) featureName = featureName.replace("~", ESCAPE)
```

Comment:
we need to fix this at the datastore/metadata level - if we fix it in tools, it won't help someone who creates a schema programmatically.